README.md

```diff
@@ -1 +1,161 @@
+    obj = s3_client.get_object(Bucket="bucket1", Key="a/b/c/obj")
+    obj2 = do_something_to_obj(obj)
+    s3_client.put_object(Body=obj2, Bucket="bucket2", Key="a/b/c/obj")
+```
+
+Suppose you want to migrate only `bucket2` to an S3-compatible storage like lakeFS or MinIO. 
+Normally, you would have to refactor your code to allow using two S3 clients instead of just one.
+Boto S3 Router allows you to do that without making any changes to the `calculate` function!
+
+## Installation
+
+### Prerequisites
+
+Boto S3 Router requires Python >= 3.6 to run.
+
+
+### Using pip 
+
+```sh
+pip install git+https://github.com/treeverse/botos3router.git
+```
+
+### Setuptools
+
+Install via [Setuptools](http://pypi.python.org/pypi/setuptools).
+
+```sh
+python setup.py install --user
+```
+
+## Basic Usage
+
+```python
+import boto3
+import botos3router as s3r
+
+# Initialize two boto S3 clients:
+s3_east = boto3.client('s3', region_name='us-east-1', signature_version='v4',)
+s3_west = boto3.client('s3', region_name='us-west-1')
+
+# Define rules for routing between the clients:
+profiles = {
+    "s3_west": {
+        "source_bucket_pattern": "example-bucket",
+        "mapped_bucket_name": "new-bucket",
+        "mapped_prefix": "test/"
+    },
+    "s3_east": {
+        "source_bucket_pattern": "example-bucket-prefix*",
+        "source_key_pattern": "a/*",
+        "mapped_prefix": "test/"
+    }
+}
+
+# Define the mapping between the profiles to the Boto clients:
+client_mapping = {"s3_west": s3_west, "s3_east": s3_east, "default":s3_east }
+
+# Initialize the client:
+client = s3r.client(client_mapping, profiles)
+
+# Use Boto S3 Router as you would use any Boto client:
+client.put_object(Bucket="bucket", Prefix="a/b/obj") # routs to s3_west, the object will be "new-bucket/test/a/b/obj
+```
+
+## Usage with [lakeFS]
+When you use Boto to access S3 and want to migrate only a subset of your data to lakeFS, Boto S3 Router allows you to use lakeFS and S3 side-by-side with minimum code changes.
+
+Consider the following code, accessing objects in two S3 buckets:
+
+```python
+import boto3
+
+s3 = boto3.client('s3')
+s3.get_object(Bucket="bucket-a", Key="test/object.txt")
+s3.get_object(Bucket="bucket-b", Key="test/object.txt")
+```
+
+Now suppose only `bucket-a` was migrated to lakeFS, and that the new repository in lakeFS is called `example-repo`. You can route only requests to this specific bucket to lakeFS, with all other requests still being routed to S3:
+
+```python
+import boto3
+import botos3router as s3r
+
+s3 = boto3.client('s3')
+lakefs = boto3.client('s3', endpoint_url='https://lakefs.example.com')
+
+profiles = {
+   "lakefs":{
+        "source_bucket_pattern": "bucket-a",
+        "mapped_prefix": "dev/"
+    }
+}
+
+s3 = s3r.client({"lakefs": lakefs, "default": s3}, profiles)
+
+# All code accessing S3 stays the same as before:
+s3.get_object(Bucket="bucket-a", Key="test/object.txt") # routes to example-repo (dev branch) in lakeFS
```

Comment:
Is 'bucket-a' the storage namespace for example-repo? 
By reading this example I can't understand what makes requests to bucket-a to be routed to example-repo. 
How does this work?

Reply:
Fixed, thanks!